pubtools/_pulp/tasks/push/command.py

```diff
@@ -144,16 +148,19 @@ def add_phase(klass, **kwargs):
             add_phase(Update)
 
             # Ensure all items are associated into the desired target repos.
-            add_phase(Associate)
+            associate_phase = add_phase(Associate)
 
             if "publish" in self.args.skip:
-                # Caller doesn't want to publish, then we just wait for prior phases
-                # to complete.
+                # Caller doesn't want to publish, then we just do post push actions
+                # for push items and wait for prior phases to complete.
+                add_phase(PostPushActions)
                 add_phase(EndPush)
 
             else:
-                # Ensure all repos are published once the desired content is present.
+                # Ensure all repos are published once the desired content is present
+                # and do any post push pushitems actions.
                 add_phase(Publish)
+                add_phase(PostPushActions)
```

Comment:
This is prone to hanging because PostPushActions has an output queue, but there is no later phase reading from it.  If the output queue fills up then the phase is going to deadlock.

For correct behavior it's required that the very last phase has no output queue.  My suggestion is that you remove the output queue from PostPushActions, and in the case above where it goes (PostPushActions => EndPush), you can just remove EndPush and leave PostPushActions as the last phase.